### ISSUE TITLE
Complete mouse piloting

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#955]
Pretty much the exact opposite of #945, allow the player to completely control their stop with only the mouse. 
This was something brought up on the Steam forums. Someone wanted to be able to move, shoot, and target enemy ships entirely with their mouse instead of using the keyboard. 
Proposed controls:
LMB: Fire
RMB: Move to position
Mouse Wheel: swap targets

Obviously you couldn't do everything on the mouse, but this would at least fill the basic controls and likely be what the person on the steam forum wanted.

**Comment by toilethinges** [comment]
As long as it'd be togglable, this'd be fine. As there's no way that I know of to make your cursor simply disappear, you'd always be going in one direction, and you'd constantly have to be piloting away from where your mouse points.

**Comment by LocalGod79** [comment]
so, if i just wanted to select a ship with the mouse, without moving to it?

**Comment by Amazinite** [comment]
That's why I suggested the RMB version, as the move to cursor version seems like it could be difficult at times.  I was just stating that the move to cursor was what they requested. Possibly a toggle on/off button for times like when you want to click to select.

**Comment by lehgogh** [comment]
@DingusShingleton good idea, but we would need to make all of these bindable (laptops don't have middle click).

**Comment by DingusShingleton** [comment]
Mine does ;) But you are right of course.

**Comment by LocalGod79** [comment]
being an accessibility issue, this would certainly be worth a look at.

the steam user, who is finding co-ordinated button pushing difficult due to nerve damage, would probably be happy with a "make ship try to point at mouse, with auto aim" toggle in the preferences, and perhaps the ability to create mouse-button key bindings.

**Comment by endless-sky** [comment]
I can think of a couple of different ways this could be done. It would take some experimentation to figure out which would be the easiest to control:

* Rely on auto-fire for primary trigger control. Have the ship always try to point toward the mouse cursor, and use left mouse for thrust and right mouse to fire secondary weapons. Maybe have auto-aim take precedence over the mouse direction unless you're actively applying thrust?

* "Tactical control" mode: rather than directly steering your ship, the mouse is just used to give the AI commands. This is already partly implemented: you can click to target a ship or planet, and double-click to board or land. Full mouse control would require at least two other commands: click to move to a certain location, and double-click a live ship to have the AI fight it.

* Mouse toggle: Allow some of the same commands as "tactical" mode above, but also let the player right click to toggle a mode where your ship is following the cursor and left mouse button controls the thrusters.

* "First-person" mode: Rotate the view so that your ship is always facing up, and have moving the mouse to the right or the left turn your ship, and moving it up or down adjust your thrust.

**Comment by Amazinite** [comment]
I'm not sure how easy it would be to have all four options available, but if there were only one then the first mode seems like it would be the best to me. Although, the "first person mode" does seem pretty interesting.

**Comment by LocalGod79** [comment]
@Amazinite would you relay M.Z.'s response to the steam user, and find out their preference order of the options from them?

i don't feel like any of us really have the perspective to choose the best implementation - their perspective will help make sure that the game stays fun, even if it's easier to play.

**Comment by Amazinite** [comment]
I've relayed it.

**Comment by Amazinite** [comment]
"You know if they put all 4 and have them set up as an option would be neet. Though I have to say I like the first one the most. The last one seems intressting but I worry that might do more damage to the game."
